server/etcdserver/v3_server.go

```diff
@@ -198,18 +196,8 @@ func (s *EtcdServer) Txn(ctx context.Context, r *pb.TxnRequest) (*pb.TxnResponse
 }
 
 func (s *EtcdServer) Compact(ctx context.Context, r *pb.CompactionRequest) (*pb.CompactionResponse, error) {
-	startTime := time.Now()
-	result, err := s.processInternalRaftRequestOnce(ctx, pb.InternalRaftRequest{Compaction: r})
 	ctx, trace := traceutil.EnsureTrace(ctx, s.Logger(), "compact")
-	if result != nil && result.Trace != nil {
```

Comment:
Can you provide more context why you remove it? Don't have enough context about the code, while most code is trivial refactor, this pops out as a removal.

Reply:
Re-added to simplify PR